lib/src/css_mirroring/util_functions.dart

```diff
@@ -0,0 +1,109 @@
+part of scissors.src.css_mirroring.bidi_css_generator;
+
+/// Makes transaction from input string.
+TextEditTransaction _makeTransaction(String inputCss, String url) =>
+    new TextEditTransaction(inputCss, new SourceFile(inputCss, url: url));
+
+///  Checks if a topLevel tree node is direction independent.
+bool _areDirIndependentDirectives(FlippableEntity<TreeNode> tnode) =>
+    (tnode.original.runtimeType == tnode.flipped.runtimeType &&
+        (tnode.original is CharsetDirective ||
+            tnode.original is FontFaceDirective ||
+            tnode.original is ImportDirective ||
+            tnode.original is NamespaceDirective));
+
+bool _areRuleSets(FlippableEntity<TreeNode> tnode) => (tnode.original is RuleSet && tnode.flipped is RuleSet);
+
+bool _areDirDependentDirectives(FlippableEntity<TreeNode> tnode) =>
+    (tnode.original.runtimeType == tnode.flipped.runtimeType &&
+        (tnode.original is MediaDirective || tnode.flipped is HostDirective));
+
+bool _areDeclarations(TreeNode a, TreeNode b) =>
+    a is Declaration && b is Declaration;
+
+bool _areDeclarationsTextEqual(Declaration a, Declaration b) =>
+    a.span.text == b.span.text;
+
+
+/// A rule can be removed if all the declarations in the rule can be removed.
+bool _isRuleRemovable(PendingRemovals removals, RuleSet rule) =>
+    removals.removalStartEndLocations.length ==
```

Comment:
As discussed previsously, I'd expose a get length and make removalStartEndLocations private